pallets/pips/src/lib.rs

```diff
@@ -528,95 +568,126 @@ decl_module! {
+        pub fn set_max_pip_skip_count(origin, new_max: SkippedCount) {
+            T::CommitteeOrigin::ensure_origin(origin)?;
+            let prev_max = MaxPipSkipCount::get();
+            MaxPipSkipCount::put(new_max);
+            Self::deposit_event(RawEvent::MaxPipSkipCountChanged(SystematicIssuers::Committee.as_id(), prev_max, new_max));
+        }
+
+        /// Change the maximum number of active PIPs before community members cannot propose anything.
+        #[weight = (150_000_000, DispatchClass::Operational, Pays::Yes)]
+        pub fn set_active_pip_limit(origin, new_max: u32) {
+            T::CommitteeOrigin::ensure_origin(origin)?;
+            let prev_max = ActivePipLimit::get();
+            ActivePipLimit::put(new_max);
+            Self::deposit_event(RawEvent::ActivePipLimitChanged(SystematicIssuers::Committee.as_id(), prev_max, new_max));
         }
 
         /// A network member creates a PIP by submitting a dispatchable which
         /// changes the network in someway. A minimum deposit is required to open a new proposal.
         ///
         /// # Arguments
+        /// * `proposer` is either a signing key or committee.
+        ///    Used to understand whether this is a committee proposal and verified against `origin`.
         /// * `proposal` a dispatchable call
-        /// * `deposit` minimum deposit value
+        /// * `deposit` minimum deposit value, which is ignored if `proposer` is a committee.
         /// * `url` a link to a website for proposal discussion
         #[weight = (1_850_000_000, DispatchClass::Operational, Pays::Yes)]
         pub fn propose(
             origin,
+            proposer: Proposer<T::AccountId>,
             proposal: Box<T::Proposal>,
             deposit: BalanceOf<T>,
             url: Option<Url>,
             description: Option<PipDescription>,
-            beneficiaries: Option<Vec<Beneficiary<T::Balance>>>
         ) -> DispatchResult {
-            let proposer = ensure_signed(origin)?;
-
-            // Pre conditions: caller must have min balance
-            ensure!(
-                deposit >= Self::min_proposal_deposit(),
-                Error::<T>::IncorrectDeposit
-            );
+            // 1. Ensure it's really the `proposer`.
+            Self::ensure_signed_by(origin, &proposer)?;
+
+            // 2. Add a deposit for community PIPs.
+            if let Proposer::Community(ref proposer) = proposer {
+                // ...but first make sure active PIP limit isn't crossed.
+                // This doesn't apply to committee PIPs.
+                // `0` is special and denotes no limit.
+                let limit = ActivePipLimit::get();
+                ensure!(limit == 0 || ActivePipCount::get() < limit, Error::<T>::TooManyActivePips);
+
+                // Pre conditions: caller must have min balance.
+                ensure!(deposit >= Self::min_proposal_deposit(), Error::<T>::IncorrectDeposit);
```

Comment:
It may be useful to enforce that for a committee PIP, the deposit is 0 (to avoid confusion).